.github/workflows/publish.yaml

```diff
@@ -4,7 +4,12 @@ on:
     branches:
       - main
 jobs:
+  ci:
+    uses: ./.github/workflows/ci.yaml # use the callable ci job to run CI checks
+
   build:
+    # CI ideally *shouldn't* fail post-merge, but you could imagine criss-crossing PRs that break tests
```

Comment:
IIRC there's a setting in the repo somewhere that requires tests not only to pass for each PR against the branch they target, but also for them to test *against the current HEAD* - we can turn that on for a saner experience